ntp-proto/src/algorithm/kalman/select.rs

```diff
@@ -0,0 +1,230 @@
+use crate::SystemConfig;
+
+use super::{config::AlgorithmConfig, PeerSnapshot};
+
+enum BoundType {
+    Start,
+    End,
+}
+
+pub(super) fn select<Index: Copy>(
```

Comment:
maybe here is not the place, but I remember that we discussed how the approach here is slightly different from the normal NTP algorithm. 

If I remember correctly, the approach here is to find the largest slice where most intervals overlap. The standard algorithm rather looks at midpoints of the intervals. 

- I think the approach here should be explained (at least a little)
- it would be useful to write down also how this is different from standard NTP (and what the tradeoffs are), but I'm not sure here is the right place for that.

Reply:
Added a short explanation that should cover most questions, and will make this a section in the white paper.